source/features/open-all-notifications.js

```diff
@@ -1,60 +1,106 @@
 import {h} from 'dom-chef';
 import select from 'select-dom';
 import delegate from 'delegate';
-import {groupButtons} from '../libs/utils';
+import * as icons from '../libs/icons';
+import {groupButtons, safeElementReady} from '../libs/utils';
 import {isNotifications} from '../libs/page-detect';
 
-function openNotifications() {
-	const urls = select.all('.unread .js-notification-target').map(el => el.href);
+const unreadNotificationsClass = '.unread .js-notification-target';
+let notificationsToOpen = [];
+
+const faceboxTrigger = (
+	<a href="#open_all_in_tabs" id="facebox-trigger" rel="facebox" style={{display: 'none'}}></a>
+);
+
+function openNotifications(unreadNotifications) {
+	const urls = unreadNotifications.map(el => el.href);
+
 	browser.runtime.sendMessage({
 		urls,
 		action: 'openAllInTabs'
 	});
-	for (const listItem of select.all('.list-group .list-group-item')) {
+
+	for (const notification of unreadNotifications) {
+		const listItem = notification.closest('.list-group-item');
 		listItem.classList.add('read');
+		listItem.classList.remove('unread');
+	}
+}
+
+function tryOpeningNotifications(container = document) {
+	const unreadNotifications = select.all(unreadNotificationsClass, container);
+
+	if (unreadNotifications.length < 5) {
+		openNotifications(unreadNotifications);
+	} else {
+		notificationsToOpen = unreadNotifications;
```

Comment:
I think that if we're going to use a global we can just have `openNotifications` only read the global so line 31 can just be `notificationsToOpen = select.all(unreadNotificationsClass, container);`